torchtune/modules/peft/lora.py

```diff
@@ -32,6 +33,8 @@ class LoRALinear(nn.Module, AdapterModule):
         rank (int): rank of the low-rank approximation
         alpha (float): scaling factor for the low-rank approximation
         dropout (float): dropout probability. Default: 0.0
+        use_dora (bool): whether to use DORA (weight-Decomposed Low-Rank Adaptation).
+            Default: False
```

Comment:
how do we want to expose DoRA? I see following the pattern of how we enabled QLoRA and just passing in a use_dora flag. Curious about the tradeoffs compared to a `DoraLinear` layer though. @ebsmothers any thoughts?